mautic/mautic
# Add support for custom timeframes with campaign lead field conditions

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | y
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

This PR removes the +/- 2-29 days for the campaign contact field condition in favor of choosing a custom interval. 

#### Steps to test this PR:
1. Apply the PR and clear browser's cache to refresh javascript files
2. Create a campaign and add a contact field value condition
3. Choose a date or date/time field then the `date` operator. Choose "Custom" from the value dropdown. An input and a select box should appear. Choose a different value from the dropdown. The input and select box should disappear. Choose Custom again then choose a different operator. The input/select should disappear. Now choose date and then custom again.
4. Enter different combinations of numbers and units (day, minute, etc). Using the browser console, watch the value of the Custom option which should change to PHP DateInterval formats as you change the values of custom date.
5. Save the condition then reopen. The input/select should repopulate with whatever was chosen before. 
6. Executing the campaign should continue to function as it did before.
